src/auth/auth.service.ts

```diff
@@ -0,0 +1,49 @@
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import * as bcrypt from 'bcrypt';
+import { User as UserInterface } from './regiter.interface';
+import { InjectRepository } from '@nestjs/typeorm';
+import User from 'src/entities/User';
+import { Repository } from 'typeorm';
+@Injectable()
+export class AuthService {
+  constructor(
+    private jwtService: JwtService,
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  async validateUser(email: string, pass: string) {
+    const user = await this.userRepository.findOne({ email });
+    const comparePassword = await bcrypt.compare(pass, user.password);
+    if (comparePassword) {
+      const { password, ...email_and_id } = user;
+      return email_and_id;
+    }
+    return null;
+  }
+  async registerUser(body: UserInterface) {
```

Comment:
```suggestion
  async registerUser(data: UserRegisterInterface hoặc UserRegisterDto) {
```

Với function với mục đích riêng thế này cần viết interface đúng mục đích, chỉ chứa các field vừa đủ